### ISSUE TITLE
Problem of training when changing to GRAB dataset

### ISSUE THREAD

**Issue by bb0928** [ManipTrans/ManipTrans#22]
Thank you for your open-source work on ManipTrans—I’ve been greatly inspired by it.

Following the dataset replacement method mentioned in your GitHub README, I switched the dataset to GRAB. However, when I tried to train the residual policy on a cubesmall_inspect sequence (~770 frames), I noticed that the episode length stopped increasing beyond 250 frames (which is roughly the point where the hand makes contact with the object). At the same time, the reward also plateaued around 1500.

When I tested the saved checkpoint, I confirmed that the policy indeed triggered failed_execute at around 250 frames. After manually disabling the failed_execute mechanism to observe the subsequent frames, I found that the hand briefly grasped the object but then lost control, causing the object to drop.

Therefore, I am wondering if this is due to insufficient training iterations, or could there be another underlying issue?

Many Thanks!

https://github.com/user-attachments/assets/3b411802-2912-442d-ae56-44b241cc94ff

**Comment by KailinLi** [comment]
Hi, thanks so much for trying ManipTrans and sharing the details.

It’s hard to diagnose from a short clip, but here are the most probable causes (highest to lowest) that you can double-check:

1. **Too few training epochs** – the default 100 works for simple tasks like *rotate_mouse*. For *cubesmall_inspect* try 500-1500 epochs.  
2. **Frame-rate mismatch** – raw GRAB is 120 fps, whereas our code assumes 60 fps. Please down-sample accordingly.  
3. **Make sure `randomStateInit=true`** during training.  
4. **Run `mano2dexhand.py` on the new GRAB sequence** with a sufficiently large `--iters` value.  
5. **Adjust the learning rate**.

If the issue persists, feel free to share more details and we’ll be happy to dig deeper.

**Comment by bb0928** [comment]
Thanks for your reply and suggestions! I followed 1,2,3 and my problem have solved.

https://github.com/user-attachments/assets/b8090314-ce47-4515-8596-5d7aec9cb90c

**Comment by goodfor7** [comment]
Hi!

I'm also trying changing to GRAB dataset right now. But I'm having a problem with the hand not making contact with the object when mano2dexhand, as shown below.

Could you please take a look at the problem?

<img width="1215" height="1059" alt="Image" src="https://github.com/user-attachments/assets/72a9e55b-b6a1-4b05-bf2d-1b53a0b171c6" />
